lib/IRGen/GenDecl.cpp

```diff
@@ -1768,6 +1769,7 @@ Address IRGenModule::getAddrOfSILGlobalVariable(SILGlobalVariable *var,
     storageType = Layout->getType();
     fixedSize = Layout->getSize();
     fixedAlignment = Layout->getAlignment();
+    castStorageToType = cast<FixedTypeInfo>(ti).getStorageType();
```

Comment:
Maybe I missed something in the logic here. But why did you introduce a new variable 'castStorageToType' and not just assign 'storageType' here?

Reply:
It is used to create the global variable.

Here:
https://github.com/apple/swift/pull/19397/files#diff-62e4305d226cedd49d95b3dff4f8b7b8R1831

Reply:
Ah, I see